tests/openedx_tagging/core/tagging/test_views.py

```diff
@@ -0,0 +1,299 @@
+"""
+Tests tagging rest api views
+"""
+
+import ddt
+from django.contrib.auth import get_user_model
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from openedx_tagging.core.tagging.models import Taxonomy
+
+User = get_user_model()
+
+
+
+@ddt.ddt
+class TestTaxonomyViewSet(APITestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.user = User.objects.create(
+            username="user",
+            email="user@example.com",
+        )
+
+        self.staff = User.objects.create(
+            username="staff",
+            email="staff@example.com",
+            is_staff=True,
+        )
+
+    @ddt.data(
+        (None, status.HTTP_200_OK),
+        (1, status.HTTP_200_OK),
+        (0, status.HTTP_200_OK),
+        (True, status.HTTP_200_OK),
+        (False, status.HTTP_200_OK),
+        ("True", status.HTTP_200_OK),
+        ("False", status.HTTP_200_OK),
+        ("1", status.HTTP_200_OK),
+        ("0", status.HTTP_200_OK),
+        (2, status.HTTP_400_BAD_REQUEST),
+        ("invalid", status.HTTP_400_BAD_REQUEST),
+    )
+    @ddt.unpack
+    def test_list_taxonomy_queryparams(self, enabled, expected_status):
+        url = reverse("oel_tagging:taxonomy-list")
+        Taxonomy.objects.create(name="Taxonomy enabled 1", enabled=True).save()
+        Taxonomy.objects.create(name="Taxonomy enabled 2", enabled=True).save()
+        Taxonomy.objects.create(name="Taxonomy disabled", enabled=False).save()
+
+        self.client.force_authenticate(user=self.staff)
+        if enabled is not None:
+            response = self.client.get(url, {"enabled": enabled})
+        else:
+            response = self.client.get(url)
+        self.assertEqual(response.status_code, expected_status)
```

Comment:
style nit for all tests: I've been using simpler asserts:
```suggestion
        assert response.status_code == expected_status
```